docs/encryption.rst

```diff
@@ -0,0 +1,77 @@
+Encrypted communication
+========================
+
+Merlin supports encrypting the packets send over the Merlin TCP protocol.
+Merlin uses libsodiums crypo_box. For more information on the encryption please
+refer to their `documentation
+<https://doc.libsodium.org/public-key_cryptography/authenticated_encryption>`_.
+
+Note that the Merlin packets body is encrypted, the header remain unencrypted.
+
+
+Configuring encryption
+-----------------------
+
+In this example we'll set up encryption between two peers, *peer01*, and *peer02*.
+
+Generating encryption keys
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To setup encryption, we first need to generating a encryption key pair consisting
+of a private key and a public key.
+
+To do so we can make sure of the ``mon merlinkey`` command.
+
+.. code-block:: none
+
+   [root@peer01 ~]# mon merlinkey generate --write
+   [root@peer02 ~]# mon merlinkey generate --write
+
+The above command does the following:
+
+1. Generates an encryption key pair and saves it at ``/etc/merlin/`` by default
+2. Adds the ``ipc_privatekey`` setting to ``merlin.conf``
+
+You can omit ``--write`` flag if you do not write the change to ``merlin.conf``.
+
+We now need to copy the public keys, to the other node. Note the private key
+should never be shared! We can use rsync over ssh to copy the relevant public
+key.
+
+.. code-block:: none
+
+   [root@peer01 ~]# rsync -a /etc/merlin/key.pub peer02:/etc/merlin/peer1.pub
```

Comment:
For a single file, perhaps `scp /etc/merlin/key.pub peer02:/etc/merlin/peer1.pub` makes for a simpler example..?